pulsar-broker/src/main/java/org/apache/pulsar/broker/ManagedLedgerClientFactory.java

```diff
@@ -154,7 +155,7 @@ public void close() throws IOException {
                 // factory, however that might be introducing more unknowns.
                 log.warn("Encountered exceptions on closing bookkeeper client", ree);
             }
-            if (bkEnsemblePolicyToBkClientMap != null) {
+            if (!bkEnsemblePolicyToBkClientMap.isEmpty()) {
```

Comment:
Iterate over an empty map doesn't produce error. So from my style preference, simply remove the if guard is better.

And the original code doesn't produce error also.

Closed as the same reason commented at https://github.com/apache/pulsar/pull/18575#issuecomment-1324754681.

Please stop picking and random "fixing" such code snippet. See also https://lkml.org/lkml/2021/6/18/153.